### DISCUSSION TITLE
[security] "zip bomb" protection?

### DISCUSSION THREAD

**Discussion by lquerel** [klauspost/compress#727] [Q&A] [answered]
Is your ZSTD implementation protected against problems similar to "[zip bomb](https://en.wikipedia.org/wiki/Zip_bomb)"? If yes do you have some documentation or code to point to? Thanks.

**Comment by klauspost** [comment, marked as answer]
[WithDecoderMaxMemory](https://pkg.go.dev/github.com/klauspost/compress@v1.15.14/zstd#WithDecoderMaxMemory) or [WithDecodeAllCapLimit](https://pkg.go.dev/github.com/klauspost/compress@v1.15.14/zstd#WithDecodeAllCapLimit) and supply a destination buffer with the capacity you don't want to exceed.

For streams you can set [WithDecoderMaxWindow](https://pkg.go.dev/github.com/klauspost/compress@v1.15.14/zstd#WithDecoderMaxWindow), if 512MB per decode is a problem. This encoder will not automatically select any window size bigger than 32MB (with SpeedBestCompression), but other implementations may have different limits.

**Reply by lquerel** [reply]
Thanks for your reply.
